aquadoggo/src/graphql/replication/single_entry_and_payload.rs

```diff
@@ -0,0 +1,53 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use async_graphql::*;
+
+use crate::db::stores::StorageEntry;
+use crate::graphql::Context as GraphQLContext;
+
+use super::payload::Payload;
+use super::Entry;
+
+/// A p2panda entry with optional payload and the collection of skiplinks required to verify it.
+#[derive(Debug)]
+pub struct SingleEntryAndPayload {
+    pub entry: Entry,
+    pub payload: Option<Payload>,
+}
+
+#[Object]
+impl SingleEntryAndPayload {
+    /// The entry
+    async fn entry(&self) -> &Entry {
+        &self.entry
+    }
+
+    /// The payload
+    async fn payload(&self) -> Option<&Payload> {
+        self.payload.as_ref()
+    }
+
+    /// Get all the skiplinks for this entry that are required to verify the entry is valid
+    async fn skiplinks<'a>(&self, ctx: &Context<'a>) -> Result<Vec<Entry>> {
```

Comment:
I think `certificatePool` might be a better name for this.

Reply:
I don't really like the naming `certificatePool`. Like, what about is it a pool? A pool for me implies that it is unordered. Just my two cents tho.

I've renamed it to certificate pool though.